dashboard/lib/services/child_account.rb

```diff
@@ -0,0 +1,23 @@
+class Services::ChildAccount
+  # Sets the child's account to a lock_out state according to our Child Account
+  # Policy.
+  def self.lock_out(user)
+    return unless user
+    # Verify the account has not already started the lock out process.
+    return if user.child_account_compliance_state
```

Comment:
Not a blocking comment, but what's the reasoning behind not updating if `child_account_compliance_state` is present? In the future, could we encounter a scenario where we actually do want to transition an account from an existing state into a locked out state?

Reply:
Currently, the expected state changes of a user are only in one direction, so I think it's important to enforce that via the code. I have seen programming technique where I could actually define a finite state machine which clearly defined states and the functions which move between states, but I want to leave that as an exercise for future projects.